examples/favorites_advanced/bloc_favorites_advanced/lib/feature_puppy/search/views/search_page.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:bloc_sample/feature_puppy/search/blocs/puppy_list_bloc.dart';
+import 'package:favorites_advanced_base/core.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:favorites_advanced_base/resources.dart';
+
+class SearchPage extends StatelessWidget {
+  static const searchPage = 'Search page';
+
+  @override
+  Widget build(BuildContext context) =>
+      BlocBuilder<PuppyListBloc, PuppyListState>(
+          key: const Key(Keys.puppySearchPage),
+          builder: (context, state) {
+            switch (state.status) {
+              case PuppyListStatus.loading:
+                context.read<PuppyListBloc>().add(LoadPuppyListEvent());
```

Comment:
@StanislavPrime, to improve the  code readability can we have each `case` as a method ?

Reply:
Each case was extracted as a separate method.